src/test/java/com/intel/gkl/pdhmm/IntelPDHMMUnitTest.java

```diff
@@ -29,94 +34,102 @@ public void pdhmmPerformanceTest() {
+                br.readLine(); // skip first line
+
+                int currentTestcase = 0;
+                while ((line = br.readLine()) != null) {
+                    String[] split = line.split("\t"); // Assuming the integers are space-separated
+                    byte[] alleleBases = split[0].getBytes(StandardCharsets.UTF_8);
+                    byte[] allelePDBases = ArrayUtils.toPrimitive(
+                            Arrays.stream(split[1].substring(1, split[1].length() - 1).split(","))
+                                    .map(num -> Byte.parseByte(num.trim())).toArray(Byte[]::new));
+                    byte[] readBases = split[2].getBytes(StandardCharsets.UTF_8);
+                    byte[] readQuals = SAMUtils.fastqToPhred(split[3]);
+                    byte[] readInsQuals = SAMUtils.fastqToPhred(split[4]);
+                    byte[] readDelQuals = SAMUtils.fastqToPhred(split[5]);
+                    byte[] overallGCP = SAMUtils.fastqToPhred(split[6]);
+                    double expected = Double.parseDouble(split[7]);
 
+                    // append testcase to full arrays
+                    System.arraycopy(alleleBases, 0, alleleBasesFull, currentTestcase * max_hap_length,
+                            alleleBases.length);
+                    System.arraycopy(allelePDBases, 0, allelePDBasesFull, currentTestcase * max_hap_length,
+                            allelePDBases.length);
+                    System.arraycopy(readBases, 0, readBasesFull, currentTestcase * max_read_length, readBases.length);
+                    System.arraycopy(readQuals, 0, readQualsFull, currentTestcase * max_read_length, readQuals.length);
+                    System.arraycopy(readInsQuals, 0, readInsQualsFull, currentTestcase * max_read_length,
+                            readInsQuals.length);
+                    System.arraycopy(readDelQuals, 0, readDelQualsFull, currentTestcase * max_read_length,
+                            readDelQuals.length);
+                    System.arraycopy(overallGCP, 0, overallGCPFull, currentTestcase * max_read_length,
+                            overallGCP.length);
+
+                    expectedFull[currentTestcase] = expected;
+                    hapLength[currentTestcase] = alleleBases.length;
+                    readLength[currentTestcase] = readBases.length;
+                    currentTestcase++;
+                }
+                br.close();
+
+                // Call Function
+                long start = System.nanoTime();
+                double[] actual = intelPDHMM.computePDHMM(alleleBasesFull, allelePDBasesFull,
+                        readBasesFull, readQualsFull, readInsQualsFull, readDelQualsFull,
+                        overallGCPFull, hapLength,
+                        readLength,
+                        testcase, max_hap_length, max_read_length);
+                long end = System.nanoTime();
+                System.out.println("Total Elapsed Time = " + (end - start) / 1e9);
```

Comment:
Please use `TimeUnit.NANOSECONDS.toSeconds(elapsedTime)` instead of dividing by 1e9